src/Traits/SearchResultHelper.php

```diff
@@ -42,8 +43,17 @@ protected function assignPaginationToCriteria(Criteria $criteria): void
         }
     }
 
-    protected function fetchProducts(Criteria $criteria, SalesChannelContext $context): EntitySearchResult
-    {
+    protected function fetchProducts(
+        Criteria $criteria,
+        SalesChannelContext $context,
+        ?string $query = null
+    ): EntitySearchResult {
+        // If a specific variant is searched by its product number, we want to modify the criteria,
```

Comment:
This comment explains the behavior really well! :+1:

Reply:
Maybe you want to put this comment as a docblock for the method `modifyCriteriaFromSearchParameter`, to make it more self-explanatory.